lib/src/local_working_copy.rs

```diff
@@ -1926,6 +1953,19 @@ impl TreeState {
                 continue;
             }
 
+            let executable = match &after {
+                MaterializedTreeValue::File(file) => file.executable,
+                MaterializedTreeValue::FileConflict(file) => file.executable.unwrap_or(false),
+                _ => false,
+            };
+            let get_exec_bit = || {
+                // We must get the previous exec bit from the file states and
+                // not the tree value because only the file states store the
+                // on-disk executable bit.
+                let get_prev_exec = || self.file_states().get_exec_bit(&path);
+                ExecBit::new_from_repo(executable, get_prev_exec)
+            };
```

Comment:
This might be true on Unix if the policy is `Ignore`, but it wouldn't be on Windows. Because `write_file()` creates new file, the file wouldn't retain the old exec bit. Since `set_executable()` is noop on Windows, I think the created file should be considered non-executable.

FWIW, I think it's okay to set exec bit to disk even if the policy is `Ignore`.